### PR TITLE
feat: Add `.gitignore` template

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

GoReleaser fails if git is in dirty state:
https://github.com/cloudquery/cq-source-plaid/actions/runs/4063182434/jobs/6995116392#step:6:27

This PR is based on https://github.com/github/gitignore/blob/main/Go.gitignore and adds the source binary and `dist` directory (GoReleaser output dir)

### CODE REVIEW

**Comment by hermanschaaf** [comment]
Do we maybe want to mark this as `chore` @erezrokah ?

**Comment by erezrokah** [comment]
> Do we maybe want to mark this as `chore` @erezrokah ?

I'm not sure I understand. This is a feature for the scaffold command, so when you run it, it will create `.gitignore` file for the newly generated plugin

File: scaffold/cmd/templates/source/.gitignore.tpl [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+*.dylib
+
+# Test binary, built with `go test -c`
+*.test
+
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
+
+# Dependency directories (remove the comment below to include it)
+# vendor/
```
Comment (candiduslynx) [inline scaffold/cmd/templates/source/.gitignore.tpl:15]: need to ignore vendor

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: scaffold/cmd/templates/source/.gitignore.tpl (hunk 1 above)
Comment (erezrokah) [reply scaffold/cmd/templates/source/.gitignore.tpl:15]: Can you please provide more context on your comment?
This is based on the GitHub template https://github.com/github/gitignore/blob/4488915eec0b3a45b5c63ead28f286819c0917de/Go.gitignore#L18, and to my understanding subjective based on the use case.

Also should this be blocking for this PR?

**Comment by hermanschaaf** [comment]
> I'm not sure I understand. This is a feature for the scaffold command, so when you run it, it will create .gitignore file for the newly generated plugin (it's not adding a .gitignore for the Scaffold module itself)

Ah sorry, I misunderstood! Makes sense 👍

**Review by hermanschaaf — APPROVED** [review]

File: scaffold/cmd/templates/source/.gitignore.tpl (hunk 1 above)
Comment (erezrokah) [reply scaffold/cmd/templates/source/.gitignore.tpl:15]: I'll go ahead a merge this and we can do a follow up PR if it makes sense

Comment (candiduslynx) [reply scaffold/cmd/templates/source/.gitignore.tpl:15]: Basically not ignoring vendor means that you commit the dependencies into your repo. It's not the case here.

**Review by candiduslynx — APPROVED** [review]
I'm approving, but I really think we should add vendor to the gitignore as [here](https://github.com/cloudquery/cloudquery/blob/main/.gitignore#L11)

File: scaffold/cmd/templates/source/.gitignore.tpl (hunk 1 above)
Comment (erezrokah) [reply scaffold/cmd/templates/source/.gitignore.tpl:15]: Got it, I think `vendor` is subjective (based on a quick research) as if you want to guarantee future reproducible builds you'll have to commit the vendor directory.
I tend to go with what ever the default is in GitHub mostly based on trust 🙃

**Pull request merged**